### DISCUSSION TITLE
Allow to copy and move logged foods from past to today (Or any day)

### DISCUSSION THREAD

**Discussion by gramapp** [jlai/fitness-dashboard#27] [Ideas]
In fitbit dash board a user could click on any logged food item, that would take him/her to nutritional information page where this food could be logged to any day and any meal. One could select day and meal before pressing log. Going to nutritional page is still an important function that I requested else where, but I think things can be improved on what fit bit used to have, read on…

Can the move dialog box also have a date where to move item to, with default to current date being viewed?
Can a "copy" button be added that will bring up a dialog box like move, but this time items are copied instead of moved?
Can a delete, move and create meal buttons be moved to the top of the page, just under the date. They are not visible w/o scrolling if there are a lot of entries?

**Comment by jlai** [comment]
Unfortunately the Fitbit API doesn't give access to nutrition info except for custom (user-created foods) which limits the value of a nutrition page. Probably they license the database from somewhere else and can't make it available to the API. Even something as simple as showing the calories before logging is problematic because it doesn't give us access to the unit conversions, e.g. if "1 banana" has 105 calories and the user eats a 120g banana, we let the server do the calculation because we don't have the info.

I am thinking though about a tab that shows the foods in a list instead of a drop-down so it's quicker to add frequently-eaten foods. Maybe with some way to assign them to categories (just in your browser, not saved to Fitbit since they don't have that).

I think keeping the management buttons at the bottom may be better, since they should be less common actions. [This page](https://ux.stackexchange.com/questions/131693/table-bulk-action-placement-guidance) discusses some of the UX tradeoffs. Moving the buttons to the top takes up screen space when you just want to view the list, while if you have a lot of foods you'll probably need to scroll anyways. In terms of discoverability, users should be aware of it because they'll start the day with no foods in the list and see the buttons.

Move and copy to another day could be interesting to have.

**Reply by gramapp** [reply]
Great, I'll be waiting for move and copy buttons. Thank you!

**Reply by jlai** [reply]
You've probably already seen the new copy button. For now, probably going to skip adding moving between days since you can do it with copy + delete and logging something on the wrong day is hopefully something that shouldn't happen super often.

**Reply by gramapp** [reply]
oh no manual labor, copy + delete, I can't bear it. JK

Thank you for the copy button, just used it. Excellent.

Move button would be nice, when you get a chance.
